### ISSUE TITLE
Support for hashing tricks

### ISSUE THREAD

**Issue by generall** [pytorch/text#281]
Currently it is nearly impossible to override logic which maps token into integer,
`numericalize` function contains a lot of other code. It is also impossible to substitute `_default_unk_index` which assign numbers for OOV tokens. The only solution I see so far is to define custom preprocessing function to remap OOV tokens to some pool of special tokens, which looks like an ugly solution.

**Comment by mttk** [comment]
So essentially, you'd like to be able to provide your own word -> int function, which would then be used for numericalization, right?

A hack is to replace stoi / itos with your custom ones in the vocab after the vocab is already built (so, after numericalization), but I agree that this should be more accessible.

**Comment by generall** [comment]
Yes, my point is to make `Vocab` more hackable. For example replace access to `stoi` and `itos` to some functions. In this case it would be easy to inherit from `Vocab` and define custom implementation for this methods.

**Comment by mttk** [comment]
Got it. I agree this should be changed, but it will require some code restructuring. Not sure on the ETA.
